homeassistant/components/syncthru/sensor.py

```diff
@@ -63,34 +75,23 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
         supp_drum = printer.drum_status(filter_supported=True)
         supp_tray = printer.input_tray_status(filter_supported=True)
         supp_output_tray = printer.output_tray_status()
-    except ValueError:
-        # if an exception is thrown, printer does not support syncthru
-        # and should not be set up
-        # If the printer was discovered automatically, no warning or error
-        # should be issued and printer should not be set up
-        if discovery_info is not None:
-            _LOGGER.info("Samsung printer at %s does not support SyncThru", host)
-            return
-        # Otherwise, emulate printer that supports everything
-        supp_toner = TONER_COLORS
-        supp_drum = DRUM_COLORS
-        supp_tray = TRAYS
-        supp_output_tray = OUTPUT_TRAYS
+    except ValueError as ex:
+        raise SyncThruNotSupported from ex
+    else:
+        if printer.is_unknown_state():
+            raise PlatformNotReady
```

Comment:
raise `ConfigEntryNotReady`. Although it might be cleaner to do that in the component (`__init__.py`) already, in case you want to add more platforms.

Reply:
I'd really like to keep this to the bare minimum and not extend to possibility to add other platforms, will leave that to others. This change is already taking more time than I actually have available for it.

But as this is platform's async_setup_entry, I think the only properly handled error here is `PlatformNotReady`. If I change that to `ConfigEntryNotReady`, this is the result for an offline printer on startup:
```
2020-07-04 07:29:11 ERROR (MainThread) [homeassistant.components.sensor] Error while setting up syncthru platform for sensor
Traceback (most recent call last):
  File "/home/scop/Documents/code/home-assistant/homeassistant/helpers/entity_platform.py", line 178, in _async_setup_platform
    await asyncio.wait_for(asyncio.shield(task), SLOW_SETUP_MAX_WAIT)
  File "/home/scop/.pyenv/versions/3.7.1/lib/python3.7/asyncio/tasks.py", line 416, in wait_for
    return fut.result()
  File "/home/scop/Documents/code/home-assistant/homeassistant/components/syncthru/sensor.py", line 82, in async_setup_entry
    raise ConfigEntryNotReady
homeassistant.exceptions.ConfigEntryNotReady
```
...whereas as-is with `PlatformNotReady`:
```
2020-07-04 07:32:13 WARNING (MainThread) [homeassistant.components.sensor] Platform syncthru not ready yet. Retrying in 30 seconds.
```
...and the appropriate retries happen.

Reply:
oh interesting. Didn't realize we used that exception for entry platform setup. Ok :+1: